website/common/script/libs/getItemInfo.js

```diff
@@ -40,10 +39,15 @@ function getDefaultGearProps (item, language) {
     per: item.per,
     con: item.con,
     klass: item.klass,
+    event: item.event,
   };
 }
 
-module.exports = function getItemInfo (user, type, item, language = 'en') {
+module.exports = function getItemInfo (user, type, item, officialPinnedItems, language = 'en') {
```

Comment:
why pass the official pinned items manually when you can get them with a call?

Reply:
I just thought that `getItemInfo` is called sometimes for a lot of items, it would be bad for the performance to recreate that array each time for each item

Reply:
👍

Reply:
Actually I just noticed that a few lines below if you do not pass the official pinned items they're calculated here so it doesn't make any difference to pass them manually (lines 47-49)

Reply:
yes some calls to `getItemInfo` don't add the officialPinned-Array (which only needed for performance reasons on multiple items), so if the parameter `officialPinnedItems` is undefined the items get calculated as a fallback